kinto_client/importer.py

```diff
@@ -0,0 +1,230 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+import argparse
+import getpass
+import logging
+import os
+from six import iteritems
+
+from kinto_client import Client
+from kinto_client.exceptions import KintoException
+
+DEFAULT_KINTO_SERVER = 'http://localhost:8888/v1'
+DEFAULT_USER_NAME = 'admin'
+
+COMMAND = 25
+logging.addLevelName(COMMAND, 'COMMAND')
+logging.basicConfig(level=COMMAND, format="%(message)s")
+logger = logging.getLogger(__file__)
+
+
+def same_record(fields, one, two):
+    for key in fields:
+        if one.get(key) != two.get(key):
+            return False
+    return True
+
+
+class KintoImporter(object):
+    set_all = False
+    bucket_permissions = None
+    collection_permissions = None
+
+    def __init__(self, *args, **kwargs):
+        self._local_records = None
+        self._remote_records = None
+
+        parser = self.configure_parser(*args, **kwargs)
+        self.args = self.get_arguments(parser)
+        self.setup_logger(logger)
+        self.setup_local_client()
+        self.setup_remote_client()
+
+    def configure_parser(self, parser=None, *args, **kwargs):
+        """Return an argparse.ArgumentParser pre-configured object."""
+
+        default = bool(kwargs.pop('set_all', self.set_all))
+        remote_server = bool(kwargs.pop('remote_server', default))
```

Comment:
I am popping the value to remove it from ``kwargs`` before passing ``kwargs`` to ArgumentParser